Common subdirectories: src/rcutils_orig/.git and src/rcutils/.git
diff -u src/rcutils_orig/CMakeLists.txt src/rcutils/CMakeLists.txt
--- src/rcutils_orig/CMakeLists.txt	2020-02-20 16:35:01.472255035 +0000
+++ src/rcutils/CMakeLists.txt	2020-02-20 16:37:41.532232214 +0000
@@ -11,6 +11,8 @@
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
+include(CheckLibraryExists)
+
 find_package(ament_cmake_python REQUIRED)
 find_package(ament_cmake_ros REQUIRED)
 
@@ -121,6 +123,8 @@
 
   find_package(launch_testing_ament_cmake REQUIRED)
 
+  check_library_exists(atomic __atomic_load_8 "" HAVE_LIBATOMICS)
+
   if(ament_cmake_cppcheck_FOUND)
     ament_cppcheck(
       TESTNAME "cppcheck_logging_macros"
@@ -214,6 +218,10 @@
       LANGUAGE C
   )
   target_link_libraries(test_atomics_executable ${PROJECT_NAME})
+  if(HAVE_LIBATOMICS)
+    target_link_libraries(test_atomics_executable atomic)
+    ament_export_libraries(atomic)
+  endif()
   add_test(NAME test_atomics COMMAND test_atomics_executable)
 
   rcutils_custom_add_gmock(test_error_handling test/test_error_handling.cpp
Common subdirectories: src/rcutils_orig/include and src/rcutils/include
Common subdirectories: src/rcutils_orig/rcutils and src/rcutils/rcutils
Common subdirectories: src/rcutils_orig/resource and src/rcutils/resource
Common subdirectories: src/rcutils_orig/src and src/rcutils/src
Common subdirectories: src/rcutils_orig/test and src/rcutils/test
